src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -64,10 +64,51 @@ void parallel_reduce (const TeamMember& team,
 
    result = local_tmp;
   } else {
-    Kokkos::parallel_reduce(Kokkos::TeamThreadRange(team, begin, end), lambda, result);
+    Kokkos::parallel_reduce(loop_boundaries, lambda, result);
   }
 }
 
+template <bool Serialize, int TotalSize, int VectorSize, typename TeamMember, typename InputProvider, typename ValueType>
+static KOKKOS_INLINE_FUNCTION
+void view_reduction (const TeamMember& team,
+                     const int& begin,
```

Comment:
If you pass TotalSize as template parameter, do you really need begin/end? My guess is 'no'.

However, I'm wondering if this impl is too restrictive. In particular, one needs to know TotalSize at compile time. For SCREAM/HOMME this is probably the case, but considering that you are not using TotalSize inside constexpr's or template parameters, we could probably make it a runtime parameter, to make the utility more flexible.

I would change the signature to `view_reduction(team, totalSize, input, result)`, with no total size parameter.

Reply:
@bartgol , the motivation for [begin,end) is that it's common in P3 and therefore likely in other parameterizations to reduce over just part of a column, with range including neither the top nor bottom. It would be useful for this function to handle the masks against [0,TotalSize).

I agree TotalSize should be a run-time, not compile-time, parameter.